cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java

```diff
@@ -76,19 +79,20 @@
   private final Supplier<EndpointStrategy> endpointStrategySupplier;
   private final Id.Namespace namespaceId;
   private final HttpRequestConfig httpRequestConfig;
+  private final boolean securityEnabled;
 
   DatasetServiceClient(final DiscoveryServiceClient discoveryClient, Id.Namespace namespaceId,
-                              CConfiguration cConf) {
+                       CConfiguration cConf) {
     this.endpointStrategySupplier = Suppliers.memoize(new Supplier<EndpointStrategy>() {
       @Override
       public EndpointStrategy get() {
         return new RandomEndpointStrategy(discoveryClient.discover(Constants.Service.DATASET_MANAGER));
       }
     });
     this.namespaceId = namespaceId;
-
     int httpTimeoutMs = cConf.getInt(Constants.HTTP_CLIENT_TIMEOUT_MS);
     this.httpRequestConfig = new HttpRequestConfig(httpTimeoutMs, httpTimeoutMs);
+    this.securityEnabled = cConf.getBoolean(Constants.Security.ENABLED);
```

Comment:
Nope, however, this one is used to check if we should send the ``CDAP-UserId`` header with the request to the dataset service (that userid will only be available in the ``SecurityRequestContext`` class if perimeter security is enabled). The dataset service will then check for authorization for this user, if a different parameter - ``security.authorization.enabled`` is set.